cli/src/gas/gas_add_budget.rs

```diff
@@ -0,0 +1,86 @@
+use {
+    crate::{command_title, display::json_output, loading, prelude::*, sui::*},
+    nexus_sdk::transactions::gas,
+};
+
+/// Upload `coin` as a gas budget for the Nexus workflow.
+pub(crate) async fn add_gas_budget(
+    coin: sui::ObjectID,
+    sui_gas_coin: Option<sui::ObjectID>,
+    sui_gas_budget: u64,
+) -> AnyResult<(), NexusCliError> {
+    command_title!("Adding '{coin}' as gas budget for Nexus");
+
+    // Load CLI configuration.
+    let conf = CliConf::load().await.unwrap_or_else(|_| CliConf::default());
+
+    // Nexus objects must be present in the configuration.
+    let NexusObjects {
+        workflow_pkg_id,
+        gas_service,
+        ..
+    } = get_nexus_objects(&conf)?;
+
+    // Create wallet context, Sui client and find the active address.
+    let mut wallet = create_wallet_context(&conf.sui.wallet_path, conf.sui.net).await?;
+    let sui = build_sui_client(&conf.sui).await?;
+
+    let address = match wallet.active_address() {
+        Ok(address) => address,
+        Err(e) => {
+            return Err(NexusCliError::Any(e));
+        }
+    };
+
+    // Fetch gas coin object.
+    let gas_coin = fetch_gas_coin(&sui, conf.sui.net, address, sui_gas_coin).await?;
+
+    // Fetch budget coin.
+    let coin = fetch_object_by_id(&sui, coin).await?;
```

Comment:
It's better to avoid variable shadowing, which can lead to confusion when reading the function:

```suggestion
    let budget_coin = fetch_object_by_id(&sui, coin).await?;
```